cdap-api/src/main/java/co/cask/cdap/api/data/stream/StreamBatchWriter.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.data.stream;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+/**
+ * Used to write to Stream in Batch mode. Once closed, a new {@link StreamBatchWriter} needs to be created to do
+ * another set of batch writes.
+ */
+public interface StreamBatchWriter extends Closeable {
```

Comment:
Actually, it'll be good for this interface extends from `WritableByteChannel` (which is almost the same as this interface right now). http://docs.oracle.com/javase/6/docs/api/java/nio/channels/WritableByteChannel.html